docs/extensibility/snippets/MailDevResource/MailKit.Client/MailKitClientFactory.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Net;
+using MailKit.Net.Smtp;
+
+namespace MailKit.Client;
+
+/// <summary>
+/// A factory for creating <see cref="ISmtpClient"/> instances
+/// given a <paramref name="smtpUri"/> (and optional <paramref name="credentials"/>).
+/// </summary>
+/// <param name="smtpUri">The <see cref="Uri"/> for the SMTP server</param>
+/// <param name="credentials">
+/// The optional <see cref="ICredentials"/> used to authenticate to the SMTP server
+/// </param>
+public sealed class MailKitClientFactory(Uri smtpUri, ICredentials? credentials = null)
+{
+    /// <summary>
+    /// Gets an <see cref="ISmtpClient"/> instance in the connected state
+    /// (and that's been authenticated if configured).
+    /// </summary>
+    /// <param name="cancellationToken">Used to abort client creation and connection.</param>
+    /// <returns>A connected (and authenticated) <see cref="ISmtpClient"/> instance.</returns>
+    /// <remarks>
+    /// Since both the connection and authentication are considered expensive operations,
+    /// the <see cref="ISmtpClient"/> returned is intended to be used for the duration of a request
+    /// (registered as 'Scoped') and shouldn't be disposed of and disconnect shouldn't be called.
```

Comment:
Who does actually dispose of it? (if it is disposable)

Reply:
Great catch! No one...how should I handle that? I want consumers to either be able to use one that's connected and then they can send as one or more message(s). But then should I have them dispose?

Reply:
The fix for this and https://github.com/dotnet/docs-aspire/pull/1355#discussion_r1678197474 might be to cache the client on the Factory, and dispose of it in a Dispose method on the Factory.

Reply:
Yes, thank you!